unit-tests/live/options/test-timestamp-domain.py

```diff
@@ -0,0 +1,117 @@
+# License: Apache 2.0. See LICENSE file in root directory.
+# Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+# test:device D400*
+
+import time
+import pyrealsense2 as rs
+
+from rspy import test
+
+
+def get_connected_device():
+    """
+    Get first connected device
+    :return: connected device
+    """
+    dev = None
+    try:
+        context = rs.context()
+        dev = context.devices[0]
+    except Exception as ex:
+        print('Exception: Failed to find connected device. ', str(ex))
+    return dev
+
+
+def close_resources(sensor):
+    """
+    Stop and Close sensor.
+    :param: sensor of device
+    """
+    if len(sensor.get_active_streams()) > 0:
+        sensor.stop()
+        sensor.close()
+
+
+def start_sensor_test(sensor, global_time_enabled: int):
+    """
+    Perform depth sensor test according given global time
+    :global_time_enabled int: 1 - time is enabled, 0 - time disabled
+    """
+
+    if global_time_enabled not in [0, 1]:
+        raise ValueError(f'Invalid parameter in start depth test: {global_time_enabled}. Only 1 or 0 are valid.')
+
+    try:
+        sensor.set_option(rs.option.global_time_enabled, global_time_enabled)
+        time.sleep(2)
```

Comment:
This will sleep for each test and make this test run for at least 4 x 2 = 8 [sec], lets try to remove it and make this test faster.
Since you are using a frame queue I don't think we need a sleep at all after test.start().
After the set, the queue.wait_for_frame() will bring the new frame with the new `time_domain` IMO.
Want to be sure? wait 100 [ms] between `set` and `wait_for_frame()`, not 2 [sec]

Reply:
I tried range time sleep durations (0.1 - 0.5) and found it failing with small values.
But after all test pass without sleep.
Sleep was removed.

Reply:
If I understand right the function wait for new frame from device. 
```python
    def wait_for_frame(self, timeout_ms=5000):
        """
        wait_for_frame(self: pyrealsense2.pyrealsense2.frame_queue, timeout_ms: int = 5000) -> pyrealsense2.pyrealsense2.frame
        
        Wait until a new frame becomes available in the queue and dequeue it.
        """
        pass
```

Reply:
We need to understand what happens if you call it when the queue is full, will you get an old frame? I think yes..

Reply:
**Yes, you're right.**
We checked C++ `wait_for_frame()` and found that it perform dequeue without any addition logic.

In addition I ran simple test:
```python
    sensor.open(depth_profile)
    sensor.start(frame_queue)

    sensor.set_option(rs.option.global_time_enabled, 0)
    time.sleep(0.3)
    sensor.set_option(rs.option.global_time_enabled, 1)

    time.sleep(0.3)
    close_resources(sensor)

    while frame_queue.size() != 0:
        frame = frame_queue.wait_for_frame()
        print(frame.get_frame_timestamp_domain(), frame)
```
Output:
timestamp_domain.hardware_clock <pyrealsense2.frame Z16 #1 @692755.822000>
timestamp_domain.hardware_clock <pyrealsense2.frame Z16 #2 @692789.160000>
timestamp_domain.hardware_clock <pyrealsense2.frame Z16 #3 @692822.497000>
timestamp_domain.hardware_clock <pyrealsense2.frame Z16 #4 @692855.835000>
timestamp_domain.hardware_clock <pyrealsense2.frame Z16 #5 @692889.173000>
timestamp_domain.hardware_clock <pyrealsense2.frame Z16 #6 @692922.511000>
timestamp_domain.global_time <pyrealsense2.frame Z16 #7 @1664437130055.003662>
timestamp_domain.global_time <pyrealsense2.frame Z16 #8 @1664437130092.119141>
timestamp_domain.global_time <pyrealsense2.frame Z16 #9 @1664437130129.703613>
timestamp_domain.global_time <pyrealsense2.frame Z16 #10 @1664437130167.756104>
timestamp_domain.global_time <pyrealsense2.frame Z16 #11 @1664437130206.278809>
timestamp_domain.global_time <pyrealsense2.frame Z16 #12 @1664437130245.270508>
timestamp_domain.global_time <pyrealsense2.frame Z16 #13 @1664437130283.071533>
timestamp_domain.global_time <pyrealsense2.frame Z16 #14 @1664437130321.071777>
timestamp_domain.global_time <pyrealsense2.frame Z16 #15 @1664437130359.273682>

Process finished with exit code 0

My conclusion: the function dequeue frames from queue by sequence.

Reply:
So what is your proposal for making it work?

Reply:
I wrote `time.sleep(0.7)`  before asking new frame by `wait_for_frame()`.
In another words I wait when frames will be updated.
`rs.frame_queue(capacity=5, keep_frames=False)` As I see according `keep_frames=False` not keep frames and keep queue with newest frames.

I have a small queue = 5 and see numbers of frames that no neighbors in queue.
> Check setting global time domain: depth sensor - timestamp domain is OFF
> timestamp_domain.hardware_clock <pyrealsense2.frame Z16 #14 @680429.211000>
> Test passed
> 
> ___
> Check setting global time domain: depth sensor - timestamp domain is ON
> timestamp_domain.global_time <pyrealsense2.frame Z16 #35 @1664448249882.792236>
> Test passed
> 
> ___
> Check setting global time domain: color sensor - timestamp domain is OFF
> timestamp_domain.hardware_clock <pyrealsense2.frame RGB8 #2 @682535.026000>
> Test passed
> 
> ___
> Check setting global time domain: color sensor - timestamp domain is ON
> timestamp_domain.global_time <pyrealsense2.frame RGB8 #24 @1664448251979.369141>
> Test passed
> 
> ___
> All tests passed (8 assertions in 4 test cases)

Reply:
But I can add function that will wait until option will swapping and skip frames.
it safety realization with small waste time.

Reply:
I mean that:
![image](https://user-images.githubusercontent.com/26552140/193014498-a4da9c73-21cd-49d5-baca-d716f82e2ae0.png)
[RealSense intro.pptx](https://rsconf.intel.com/display/RealSenseSWRD/Intro+to+LibRS?preview=/87248400/87248423/RealSense%20Intro.pptx)

Reply:
Let's talk about it next week